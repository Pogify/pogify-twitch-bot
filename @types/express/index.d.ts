declare namespace Express {
  interface Request {
    twitch: {
      broadcaster_type: BroadecasterType;
      description: string;
      display_name: string;
      email?: string;
      id: string;
      login: string;
      offline_image_url: string;
      profile_image_url: string;
      type: type;
      view_count: number;
      created_at: string;
    };
  }
}

type BroadecasterType = "partner" | "affiliate" | "";
type type = "staff" | "admin" | "global_mod" | "";
