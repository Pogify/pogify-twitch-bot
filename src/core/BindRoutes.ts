import { Express } from "express";

import consoleRouter from "../routes/console";
import initializeRouter from "../routes/initialize";
import botControlRouter from "../routes/bot_control";

export default function BindRoutes(app: Express): void {
  app.use("/", consoleRouter);
  app.use("/init", initializeRouter);
  app.use("/", botControlRouter);
}
