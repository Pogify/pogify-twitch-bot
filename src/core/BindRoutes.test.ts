import BindRoutes from "./BindRoutes";
import express from "express";

test("test bindRoutes", () => {
  const app = express();

  BindRoutes(app);
  expect(app._router.stack).toMatchSnapshot();
});
