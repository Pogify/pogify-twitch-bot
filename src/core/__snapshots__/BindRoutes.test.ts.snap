// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test bindRoutes 1`] = `
Array [
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "query",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "expressInit",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "router",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "router",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/init\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "router",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
]
`;
